tests/e2e/login/purge.spec.js

```diff
@@ -1,315 +0,0 @@
-    {
-      _id: goodFormId2,
-      form: 'a-good-form-type',
-      type: 'data_record',
-      content_type: 'xml',
-      reported_date: Date.now(),
-      contact: {
-        _id: restrictedContactId,
-        parent: {
-          _id: restrictedFacilityId,
-          parent: {
-            _id: 'this-does-not-matter'
-          }
-        }
-      },
-      fields: {
-        patient_id: patientId,
-        patient_name: 'A patient',
-        some: 'data',
-      }
-    },
-    {
-      _id: badFormId2,
-      form: 'a-bad-form-type',
-      type: 'data_record',
-      content_type: 'xml',
-      reported_date: Date.now(),
-      contact: {
-        _id: restrictedContactId,
-        parent: {
-          _id: restrictedFacilityId,
-          parent: {
-            _id: 'this-does-not-matter'
-          }
-        }
-      },
-      fields: {
-        patient_id: patientId,
-        patient_name: 'A patient',
-        some: 'data',
-      }
-    }
-  ];
-
-  const purgeFn = (userCtx, contact, reports) => {
-    if (!userCtx.roles.includes('data_entry')) {
-      // wrong user type - don't purge
-      return [];
-    }
-    if (contact.type !== 'person') {
-      // report not about person - don't purge
-      return [];
-    }
-    return reports.filter(r => r.form === 'a-bad-form-type').map(r => r._id);
-  };
-
-  let originalTimeout;
-  beforeAll(async () => {
-    originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
-    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
-    await commonElements.goToMessagesNative();
-    await utils.saveDocs(initialReports.concat(initialDocs));
-    await utils.request({
-      path: `/_users/org.couchdb.user:${restrictedUserName}`,
-      method: 'PUT',
-      body: restrictedUser
-    });
-    const seq = await sentinelUtils.getCurrentSeq();
-    await utils.updateSettings({ purge: { fn: purgeFn.toString(), text_expression: 'every 1 seconds' } }, true);
-    await restartSentinel();
-    await sentinelUtils.waitForPurgeCompletion(seq);
-  });
-
-  afterAll(async () => {
-    jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout;
-    await utils.revertDb([], 'api');
-    await commonElements.goToLoginPageNative();
-    await loginPage.loginNative(auth.username, auth.password);
-    await utils.deleteUsers([restrictedUserName]);
-    await sentinelUtils.deletePurgeDbs();
-    await commonElements.calmNative();
-  });
-
-  beforeEach(utils.beforeEach);
-  afterEach(utils.afterEach);
-
-  const getPurgeLog = () => {
-    return browser.executeAsyncScript((() => {
-      const callback = arguments[arguments.length - 1];
-      const db = window.CHTCore.DB.get();
-      db.get('_local/purgelog')
-        .then(doc => callback(doc))
-        .catch(err => callback(err));
-    }));
-  };
-
-  const restartSentinel = () => utils.stopSentinel().then(() => utils.startSentinel());
-
-  it('Logging in as a restricted user with configured purge rules should not download purged docs', async () => {
```

Comment:
Don't you think there's still value in having this around? it actually tests which docs get purged, with different purge rules, while the other test does not.

Reply:
I'm not sure I follow. The purge.wdio-spec.js e2e test has one configured purge function, runs purge on the server, refreshes the browser, makes sure the right docs are purged, then changes the purge function, runs purge on the server, refreshes the browser, and makes sure additional docs are purged. What is it missing?

Reply:
The one thing it's missing is all the assertions on purge history contents. 
Maybe we could add assertions for `to_purge` after sync and add assertions for the purge history entry in this test?

Reply:
I've added the extra assertions.